outrank/task_instance_ranking.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+
+import gzip
+import logging
+import os
+from collections import Counter
+from collections import defaultdict
+from typing import Any
+
+import numpy as np
+import pandas as pd
+import tqdm
+
+from outrank.core_utils import generic_line_parser
+from outrank.core_utils import get_dataset_info
+from outrank.core_utils import get_num_of_instances
+
+try:
+    import matplotlib.pyplot as plt
+except:
+    pass
+
+
+def shannon_ent(string):
+    counts = Counter(string)
+    frequencies = ((i / len(string)) for i in counts.values())
+    return -np.sum(f * np.log2(f) for f in frequencies)
+
+
+def compute_entropy_avg(line):
+    joint_ent = 0
+    for field in line:
+        joint_ent += shannon_ent(field)
+    return joint_ent
+
+
+def score_line(line):
+    nan_prop = line.count('') / len(line)
+    out_struct = {}
+    out_struct['empty_string_prop'] = nan_prop
+    out_struct['empty_dict'] = line.count('{}') / len(line)
+    out_struct['all_empty'] = (line.count('{}') + line.count('')) / len(line)
+    out_struct['all_zero'] = line.count('0') / len(line)
+    for j in [30, 60, 100, 200, 300]:
+        out_struct[f'all_more_{j}_chars'] = len(
+            [x for x in line if len(x) > j], ) / len(line)
+    out_struct['row_entropy'] = compute_entropy_avg(line)
+    return out_struct
+
+
+def outrank_task_rank_instances(args: Any) -> None:
+
+    data_encoding = 'utf-8'
+    delimiter = '\t'
+    dataset_info = get_dataset_info(args)
+    local_pbar = tqdm.tqdm(
+        total=get_num_of_instances(dataset_info.data_path) - 1,
+        position=0,
+        disable=args.disable_tqdm == 'True',
+    )
+    local_pbar.set_description('Starting ranking computation')
+
+    file_name, file_extension = os.path.splitext(dataset_info.data_path)
```

Comment:
`file_name` is not used, maybe replace with `_`.

Reply:
sure